kernel/nativelib.ml

```diff
@@ -86,9 +86,25 @@ let call_compiler ?profile:(profile=false) ml_filename =
     else
       []
   in
+  let flambda_args =
+    if Coq_config.caml_version_nums >= [4;3;0] then
+      (* We added support for flambda-specific native compiler flags
+         in order to help compilation of OrdersEx with flambda;
+         unfortunately, the problem is deeper than that and the flags
+         don't help, see https://caml.inria.fr/mantis/view.php?id=7630
+
+         Anyways, we play safe for now, and use the native compiler
+         with -Oclassic, however it is likely that `native_compute`
+         users could benefit from tweaking here.
+      *)
+      ["-Oclassic"]
```

Comment:
Sorry, I don't know these flags yet. Does this have an effect on people not using flambda?

Reply:
Unless I am mistaken, these flags are ignored if OCaml was built without the -flambda switch, [only in compilers >= 4.03.0] My understanding is that it is safe to use the same sets of flags both in `flambda: yes` and `flambda:no`, with the caveat that in the `no` case, the flags will be noops. Thus, I decided against having yet-another-switch in Coq.

Regarding using `-Oclassic` in the native compiler, I think it may not be the best choice, as the bug tuned out to be in other part of OCaml [see mantis] . We will have to update the comment once we do the testing suggested in the OCaml's PR.

Reply:
There is the matter of stability of flambda (the more aggressively you optimize, the more likely you introduce miscompilations or have the compiler fail), but there is also the matter of the compilation-time vs. run-time tradeoff. I would guess that native-compute users are fairly sensitive to the compilation-time aspect, so I think that a default choice that tries to minimize compilation time (at a reasonable efficiency cost) is best. In particular, this is not related to flambda, but I think that using `-linscan` (the faster register allocator, available on 4.06) would be a win for native-compute.

Reply:
(Currently I think that flambda with `-Oclassic` is still sensibly slower than the non-flambda compiler for comparable compilation quality, so in that sense it is a bad deal for native-compute. But you cannot use `-flambda` for Coq and its plugins, and not use it for native-compute. Furthermore, we expect important improvements to `-Oclassic` before the 4.07 release that should make it more competitive with the non-lambda backend.)

Reply:
> I would guess that native-compute users are fairly sensitive to the compilation-time aspect

I'd dare to say that `native_compute` users will not care about compilation time, their expected runtime ought to be dominated by the actual reduction [otherwise native_compute offers little advantage]. In particular, I'd expect them to choose the most aggressive optimization options, not `-Oclassic`.

> you cannot use -flambda for Coq and its plugins, and not use it for native-compute.

Umm, I don't see why this is the case. Does `native_compute` link in any form with Coq @maximedenes  ?